### ISSUE TITLE
limit stem length on grace notes

### ISSUE THREAD

**Issue by rettinghaus** [rism-ch/verovio#419]
@craigsapp was wrong in #40 by saying:
> The stems which would otherwise end below the middle line of the staff must be extended to touch the middle line. No exceptions, and it is an extreme violation of notational conventions to not do so.

Actually there is an exception with grace notes. Grace note stems should never be longer than 5 vu (MEI virtual units). 

Verovio shows this:
![gracestemvrv](https://cloud.githubusercontent.com/assets/7693447/21772958/9b4538ea-d68d-11e6-81e9-0e5278a59eb6.png)
But it should look this way (MuseScore):
![gracestem](https://cloud.githubusercontent.com/assets/7693447/21772977/a1b5bf56-d68d-11e6-9bfe-6b7724f322df.png)

```xml
<score>
  <scoreDef xml:id="scoredef-000000094643793">
    <staffGrp xml:id="m-000000117201221">
      <staffGrp xml:id="m-000000142930896" symbol="brace" barthru="true">
        <staffDef xml:id="staffdef-000000038590390" clef.shape="G" clef.line="2" key.sig="0" meter.count="4" meter.unit="4" meter.sym="common" n="1" lines="5" />
        <staffDef xml:id="staffdef-000000133835307" clef.shape="F" clef.line="4" key.sig="0" meter.count="4" meter.unit="4" meter.sym="common" n="2" lines="5" />
      </staffGrp>
    </staffGrp>
  </scoreDef>
  <section xml:id="section-000000065838342">
    <measure xml:id="measure-000000189948540" n="1">
      <staff xml:id="staff-000000114853788" n="1">
        <layer xml:id="layer-000000101983932" n="1">
          <mRest xml:id="mrest-000000210670441" />
        </layer>
      </staff>
      <staff xml:id="staff-000000190974671" n="2">
        <layer xml:id="layer-000000174776323" n="5">
          <note xml:id="note-000000037515609" dur="8" oct="2" pname="c" grace="acc" stem.dir="up" />
          <chord xml:id="chord-000000089405053" dur="1">
            <note xml:id="note-000000139119585" oct="2" pname="c" />
            <note xml:id="note-000000134160627" oct="3" pname="c" />
          </chord>
        </layer>
      </staff>
      <slur xml:id="slur-000000035116997" startid="#note-000000037515609" endid="#chord-000000089405053" />
    </measure>
    <measure xml:id="measure-000000019015014" n="2" right="end">
      <staff xml:id="staff-000000193772054" n="1">
        <layer xml:id="layer-000000155313133" n="1">
          <mRest xml:id="mrest-000000018248990" />
        </layer>
      </staff>
      <staff xml:id="staff-000000123443746" n="2">
        <layer xml:id="layer-000000161146122" n="5">
          <note xml:id="note-000000104363871" dur="8" oct="1" pname="f" grace="acc" stem.dir="up" />
          <note xml:id="note-000000061170188" dur="4" oct="1" pname="f" stem.dir="up" />
          <rest xml:id="rest-000000021629391" dur="4" />
          <rest xml:id="rest-000000146175490" dur="2" />
        </layer>
      </staff>
      <slur xml:id="slur-000000152084297" startid="#note-000000104363871" endid="#note-000000061170188" />
    </measure>
  </section>
</score>
```

**Comment by craigsapp** [comment]
> Grace note stems should never be longer than 5 vu (MEI virtual units).

Says who?

The second grace note in the MuseScore example is particularly difficult to read.

Here is the example typeset in SCORE, where the rule is that long grace note  stems are placed one line higher/lower than the middle staff line:

<img width="703" alt="screen shot 2017-01-09 at 08 32 57" src="https://cloud.githubusercontent.com/assets/3487289/21774112/41eaa72c-d646-11e6-93c0-7248752169f7.png">

<img width="991" alt="screen shot 2017-01-09 at 08 33 25" src="https://cloud.githubusercontent.com/assets/3487289/21774126/5088791c-d646-11e6-940d-d817dad8669c.png">

**Comment by craigsapp** [comment]
I don't recall seeing any such grace notes -- typically the grace note would probably go in the opposite stem direction as the main note, especially in these situations and thus avoid the problem.  You brought it up, so *you* can provide the exemplar :-)

The lilypond rendering is readable, so not a problem like the MuseScore rendering.  In the lilypond case, the ledger lines do not get in the way of the grace note flag, and the slash also avoids interacting with the ledger lines.

However this is still not standard practice in professional typesetting.  Here is what Behind Bars says (page 126):

<img width="431" alt="screen shot 2017-01-09 at 11 31 45" src="https://cloud.githubusercontent.com/assets/3487289/21780306/e6b9d142-d65f-11e6-88f1-9ff9e7efd1d6.png">

It does not explicitly say the second staff line, but it seems to be implied.

<img width="315" alt="screen shot 2017-01-09 at 11 37 22" src="https://cloud.githubusercontent.com/assets/3487289/21780343/09df7dd4-d660-11e6-9861-e8f7deda3d33.png">

Sibelius does a modified version of SCORE, limiting the stem to the bottom/top line of the staff:

<img width="621" alt="screen shot 2017-01-09 at 11 21 29" src="https://cloud.githubusercontent.com/assets/3487289/21779854/25a18d3e-d65e-11e6-8d3d-6c1c73fc9fdb.png">

The flag/slash are mostly avoiding the ledgerlines, but I find it less readable than the SCORE example since the slash is crossing over the top ledger line.  

In other words, any visual disturbance which increases the time for the brain to recognize that there are four leger-lines for the second grace note is bad (this is the underlying universal rule in music notation, which is to optimize sight-reading efficiency).  The lilypond rendering does not violate this rule, so it is acceptable.  In my opinion Sibelius is violating this rule, but not very much.

**Comment by lpugin** [comment]
For now grace note stems are left short (e69f8765627dc29c16b2f703d29f7637f12d0da1)
